### DISCUSSION TITLE
Backport #1660 to 2.7.5 and 3.0?

### DISCUSSION THREAD

**Discussion by Bananeweizen** [eclipse-tycho/tycho#1750] [General]
I've just been checking how much slower target resolution is due to the relative path issue from #1660. I have a multi module project at work, where the difference is several minutes (!!) on my laptop. That's because
* 50 modules
* the target contains 2 further nested targets
* one of the nested targets contains 30 Maven artifacts, of those several are automatically wrapped by Tycho

That all multiplies, so Tycho basically resolves 150 targets instead of 3 and wraps some hundred Maven artifacts for no good. I've meanwhile found a workaround by exchanging `${basedir}/../somedir/my.target` with `${maven.multiModuleProjectDirectory}/somedir/my.target`, so I can live without a backport. Nevertheless there might be others affected by similar slowdowns (if they use maven artifacts in their targets). Therefore this might be a good candidate for backporting to the older versions.

**Comment by laeubi** [comment]
I have added a section about backporting to the contributing guide now:
https://github.com/eclipse-tycho/tycho/blob/master/CONTRIBUTING.md#backporting

@Bananeweizen do you like to drive the backporting steps for this (and possibly other items as well?).

**Reply by lppedd** [reply]
I would like to coordinate with @Bananeweizen to include #1625 and #1681, if possible.

**Comment by laeubi** [comment]
Just let you know that we prepare a 3.0.1 snapshot. Because there is a major problem to fix, we won't have much time and won't have a test-period, but trivial fixes can be included as of 12:00 GMT+1:00 if offered by a sucessful PR against the tycho 3.0.x branch.

**Comment by laeubi** [comment]
@Bananeweizen @lppedd we have already some fixes for 3.0.2 and probably there is an opportunity to perform a new release soon, so if you like to get your changes included it would be good to just start with that, just leave a note if you are working on a backport here should be enough for coordination.
